src/Controller/Api/AbstractController.php

```diff
@@ -34,67 +34,54 @@
     {
         $this->serializer = $serializer;
     }
 
-    /**
-     * @param string $version
-     * @return \App\Entity\MajorVersion
-     */
     protected function findMajorVersion(string $version): MajorVersion
     {
         $this->checkMajorVersionFormat($version);
-        $repo = $this->getDoctrine()->getRepository(MajorVersion::class);
-        $entity = $repo->findOneBy(['version' => $version]);
-        if (null === $entity) {
+        $majorVersion = $this->getDoctrine()->getRepository(MajorVersion::class)->findOneBy(['version' => $version]);
+        if (!$majorVersion instanceof MajorVersion) {
             throw new NotFoundHttpException('No such version.');
         }
-        return $entity;
+        return $majorVersion;
     }
 
     /**
-     * @param \Symfony\Component\Validator\Validator\ValidatorInterface $validator
      * @param $object
      */
     protected function validateObject(ValidatorInterface $validator, $object): void
     {
-        $errors = $validator->validate($object);
+        $constraintViolationList = $validator->validate($object);
 
-        if (\count($errors) > 0) {
-            $errorsString = (string)$errors;
+        if (\count($constraintViolationList) > 0) {
+            $errorsString = (string)$constraintViolationList;
             throw new BadRequestHttpException($errorsString);
         }
     }
 
     protected function mapObjects($baseObject, array $data): void
     {
-        $em = $this->getDoctrine()->getManager();
-        $metadata = $em->getMetadataFactory()->getMetadataFor(\get_class($baseObject));
-        foreach ($metadata->getFieldNames() as $field) {
+        $classMetadata = $this->getDoctrine()->getManager()->getMetadataFactory()->getMetadataFor(\get_class($baseObject));
+        foreach ($classMetadata->getFieldNames() as $field) {
             $fieldName = Inflector::tableize($field);
             if (is_array($data)) {
                 $data = $this->flat($data);
             }
             if (array_key_exists($fieldName, $data)) {
-                if (isset($metadata->fieldMappings[$field]['type'])) {
-                    switch ($metadata->fieldMappings[$field]['type']) {
-                        case 'datetime':
-                            $data[$fieldName] = new \DateTime($data[$fieldName]);
-                            break;
-                        case 'datetime_immutable':
-                            $data[$fieldName] = new \DateTimeImmutable($data[$fieldName]);
-                            break;
+                if (isset($classMetadata->fieldMappings[$field]['type'])) {
+                    if ($classMetadata->fieldMappings[$field]['type'] == 'datetime') {
+                        $data[$fieldName] = new \DateTime($data[$fieldName]);
+                    } elseif ($classMetadata->fieldMappings[$field]['type'] == 'datetime_immutable') {
+                        $data[$fieldName] = new \DateTimeImmutable($data[$fieldName]);
```

Comment:
In PHP 8 this can be a `match()` statement instead. :smile:

Reply:
Yeah, there is also a rector for this change...